src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -316,8 +324,9 @@ int halide_hexagon_remote_run(handle_t module_ptr, handle_t function,
 }
 
 int halide_hexagon_remote_poll_log(char *out, int size, int *read_size) {
+    // Read one line at a time.
     // Leave room for appending a null terminator.
-    *read_size = global_log.read(out, size - 1);
+    *read_size = global_log.read(out, size - 1, '\n');
     out[*read_size - 1] = 0;
```

Comment:
I just looked over the code. It seems to me Log::read will return 0 when there is nothing written in the log. (This is a polling call, it doesn't wait.) This means the code will write a NUL to out[-1]. If I'm reading the code right, this is gonna do something unpleasant on the stack in poll_log in hexagon_host.cpp. If there is some reason it is impossible for this call to be made when the log is empty, then it would not be doing the bad write on a regular basis, but I think it is still worth fixing.